### PR TITLE
more flexible perf test check

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
This PR separates the "last known valid perf test" commit from the "baseline speedy implementation" commit. It is important for the perf test to be meaningful that the changes between those two commits are benign, say minor API adjustments, so that the perf measurement remains meaningful.

This also adds a check on merging to master that tells Slack if the perf test has changed and the `test_sha` file needs updating. The Slack message is conditional on the current commit to avoid excessive noise.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ci/cron/test_sha [hunk 1]
```diff
@@ -0,0 +1 @@
+53b24793f4a90b777198c7e6dd4e509b4085c140
```
Comment (cocreature) [inline ci/cron/test_sha:1]: Maybe worth giving this a more descriptive file name or move it to a `perf` subdirectory?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -260,6 +260,26 @@ jobs:
...
 
+  - job: check_perf_test
+    pool:
+      name: linux-pool
+    condition: eq(variables['Build.Reason'], 'IndividualCI')
+    steps:
+    - bash: |
+        TEST_SHA=$(cat ci/cron/test_sha)
+        LAST_CHANGES=$(git log -n1 --format=%H daml-lf/scenario-interpreter/src/perf)
+        CURRENT_SHA=$(git rev-parse HEAD)
```
Comment (garyverhaegen-da) [inline azure-pipelines.yml:271]: This only runs on "individual CI", i.e. not PRs, which means
- It does not need to worry about Azure running merge commits; the current commit really is the one we're running for.
- This is not tested as part of the PR checks. 🙏 driven development.

File: ci/cron/test_sha (hunk 1 above)
Comment (garyverhaegen-da) [reply ci/cron/test_sha:1]: how does `perf_test_sha` sound?

Comment (cocreature) [reply ci/cron/test_sha:1]: Sounds great! I just want to have the `perf` part somewhere in there :slightly_smiling_face:

Comment (garyverhaegen-da) [reply ci/cron/test_sha:1]: moved `test_sha` and `compare-perf.sh` to `ci/cron/perf` folder.

**Pull request merged**